### PR TITLE
Update for juttle 0.5.0.

### PR DESCRIPTION
**Pull request by mstemm**
Copy dot files from juttle-adapter-template to switch to gulp and eslint.

Install juttle ^0.5.0 in travis instead of the latest juttle.

Add a juttleAdapterAPI element to package.json with version 0.5.0.

Make everything ES6 friendly with classes, fat arrow for loops, no
more self, let instead of var, for (item of array) instead of forEach,
etc.

Also fix everything found by eslint, mostly use of double quotes.

In FilterGmailCompiler, make the recommendations suggested in
https://github.com/juttle/juttle/wiki/Porting-Adapters-to-the-0.5.x-API:

 - no visitExpressionFilterTerm
 - rely on visitFulltextFilterTerm instead of visitSimpleFilterTerm
 - use visitField instead of visitUnaryExpression + '*'.
 - use featureNotSupported to return errors instead of throwing our own.

In the filter tests, this means creating a FilterSimplifier and
running it on the ast.

In the main index.js, validate that the config has the keys we want.

In the adapter classes, switch to a class deriving from
AdapterRead/AdapterWrite and use JuttleAdapterAPI to access
everything. Use the built-in code for validating options instead of
doing our own checks. Get rid of interactions with the scheduler.

In write, use the sql adapter's trick of keeping an ongoing promise
that is appended to as more write()s are called, and is returned in
eof().

@VladVega @demmer @davidvgalbraith @rlgomes

### CODE REVIEW

File: lib/read.js [hunk 1]
```diff
@@ -1,113 +1,77 @@
...
-            throw this.compile_error('RT-MISSING-TIME-RANGE-ERROR');
-        }
+    defaultTimeRange() {
+        return {
+            from: this.params.now,
+            to: new JuttleMoment(Infinity)
+        };
+    }
 
-        // If 'from'/'to' are present, 'last' can not be present.
```
Comment (rlgomes) [inline lib/read.js]: This check isn't done automagically for you if removing this was intended to acheive that. You can see [here](https://github.com/juttle/juttle-elastic-adapter/blob/master/lib/read.js#L58) how you have to check for from/to or last being set.

**Comment by mstemm** [comment]
Also updated the walkthrough docs to reflect changes in 0.5.0.

File: lib/read.js (hunk 1 above)
Comment (mstemm) [inline lib/read.js]: I have a unit test for this which is passing (although I noticed I need to check for false negatives), and I see https://github.com/juttle/juttle/blob/master/lib/runtime/procs/source.js#L72.

With the addition of defaultTimeRange(), it's now acceptable to not specify any of -from/-to/-last, and I just updated the tests to reflect that.

**Comment by VladVega** [comment]
Looks good to me. I especially liked the `let prop of Array` stuff. Very nice. Nice filter refactor too. I imagine we should all refactor down that path.

**Pull request merged**